src/clj/commiteth/eth/core.clj

```diff
@@ -277,36 +316,68 @@
     (eth-rpc {:method "eth_call"
               :params [{:to contract :data data} "latest"]})))
 
-(defn execute
-  [{:keys [from contract method-id gas-limit params internal-tx-id]}]
-  {:pre [(string? method-id)]}
+(defn construct-params
+  [{:keys [from contract method-id params gas-price]}]
   (let [data (apply format-call-params method-id params)
+        value (format "0x%x" 0)]
+    (cond-> {:data data
+             :from  from
+             :value value}
+      gas-price
+      (merge {:gasPrice (integer->hex gas-price)})
+      contract
+      (merge {:to contract}))))
+
+(defmulti execute (fn [arg] (offline-signing?)))
```

Comment:
Using a multimethod with boolean is a bit unusual, for readability it might be better to make the dispatch fn return some more descriptive value:

```clojure
(if (offline-signing?) :offline :whatever-youd-call-the-other-thing)
```

Alternatively you could just have something like

```clojure
(defn execute-offline ,,,)

(defn execute-bla ,,,)

(defn execute ,,,
  (case (offline-signing?)
    true (execute-offline ,,,)
    false (execute-blabla ,,,)))
```

Multimethods shine when you're trying to create an open extension mechanism, that is you want to allow other people to provide additional implementations of this functionality. In this case I'm not sure how likely it is that we will do this. 

Definitely nothing wrong with using a multimethod here though! :+1: